Sources/EmbraceCoreDataInternal/CoreDataWrapper.swift

```diff
@@ -144,25 +183,30 @@ public class CoreDataWrapper {
                 self?.context.delete(record)
```

Comment:
Reuse of `self` here can be dangerous since it's weak and could go nil at any point which would lead to data issues and we wouldn't know. Do the `guard let self else { return }`. 

This also makes me wonder if these functions should be throwing so we can handle data corruption and errors at a higher level. The new `catch` and logging is good, but we'll still lose data and not really have insight into it. This can probably wait for another PR however. Maybe we need a new feature at some point that adds observability to our SDK ;P